api/v1alpha1/edgeplaybook_types.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// EdgePlaybookSpec defines the desired state of EdgePlaybook
+type EdgePlaybookSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	DeviceSelector *metav1.LabelSelector `json:"deviceSelector,omitempty"`
+
+	// The ansible playbook command to execute
+	AnsiblePlaybookCmd *AnsiblePlaybookCmd `json:"ansiblePlaybookCmd,omitempty"`
+
+	LogCollection map[string]*LogCollectionConfig `json:"logCollection,omitempty"`
+}
+
+type AnsiblePlaybookCmd struct {
+	// username who execute the playbook
+	User string `json:"user,omitempty"`
+	// Ansible binary file
+	BinaryPath string `json:"binaryPath,omitempty"`
+	// the ansible's playbooks list to be used
+	Playbooks []Playbook `json:"playbooks,omitempty"`
+	// the ansible's playbook options
+	Options *AnsibleOptions `json:"ansibleOptions,omitempty"`
+	// the ansible's playbook privilege escalation options
+	PrivilegeEscalationOptions *PrivilegeEscalationOptions `json:"privilegeEscalationOptions,omitempty"`
+	// enum that defines which is the stdout callback method (default = 'default'). Supported stdout methods are: debug,default,dense,json,minimal,null,oneline,stderr,timer,yaml,runner-events
+	StdoutCallback string `json:"stdoutCallback,omitempty"`
```

Comment:
We will collect output and send it back as events. Why do we need it?

Reply:
My doubt here is still: do we want to be compliant **only** with fleet management?

This field specifies the format of the events sent back. (Probably I choose a bad name for it 👎🏻  )

If we want to keep the possibility for the user to receive the "standard" output of `ansible-playbook`, then the user can specify the desidered `Stdoutcallback` (`json`,`yaml`, ...).
Specifying a `Stdoutcallback` equals to `runner-events` means that the user wants results compliant with fleet-management (which means translating from `ansible-playbook` output to `runner-events`). 

The information contained in the `ansible-playbook` output are much more than those in the `runner-events`, that's why I think it may be useful.

However I can remove the `Stdoutcallback` field and always sent the result as `runner-events`.
We can also decide to remote this field as first step and integrate other output format in the future.

I decided to leave this field in this PR so we would get a chance to discuss it :)

Reply:
The thing is, stdout output to where?

Reply:
I am all for having support on the device but we can use one format and expose it either by events or as part of the status.

Reply:
I'm sorry Piotr but I can completely understand you suggestion.
I get the idea, and I'm ok with it :) , but I can't understand how to implement that.

Can you give me an example, please?

In addition, ATM the Flotta agent always translate from json to runner-events. If we want to expose one format (i.e. `default`) and `runner-events` than we should plan to change that logic in the agent and to provide (in the operator?) a way to convert from `default` to all the other possibile format (i.e. `yaml`, `dense`, `timer`, `json`, ...). Or am i completely on the wrong track?

Reply:
As far as I understand this parameter it defines format of ansible output on the device. My point is that we will process the output on the device and send it back to operator so it can be exposed via kubeapi so I am not sure why user would want different formats.

Please correct me if I misunderstood something.